src/components/Identity/index.js

```diff
@@ -26,18 +26,19 @@ const createIdentity = ({ config, logger, cookie }) => {
   const getEcid = () => cookie.get(ECID_NAMESPACE);
 
   let ecid = getEcid();
+  let responseRequested = false;
   let deferredForEcid;
 
-  const onBeforeEvent = (event, isViewStart) => {
-    if (!isViewStart) {
-      return;
+  const onBeforeEvent = event => {
+    if (!ecid && !responseRequested) {
+      // TODO: Remove; We won't need to request id syncs explicitely.
+      // This is just for demo currently.
+      event.mergeQuery({
+        idSyncs: true
+      });
+      event.expectResponse();
+      responseRequested = true;
```

Comment:
Long term, we need to figure out a way for the identity component to know when the id request fails or times out.

Reply:
Yes...and what we want to do about it when it does fail or time out...

Reply:
We have this ticket that might cover that use case: https://jira.corp.adobe.com/browse/CORE-28205